### PR TITLE
Add a callable to call each time log submission fails

### PR DESCRIPTION
**Pull request by piotrmaslanka**
So that you can detect via other mechanisms (eg. metrics) that you logging has just failed

### CODE REVIEW

**Comment by tintoy** [comment]
Thanks - I like this idea! 🙂

Will have a look at the PR when I get home from work.

**Comment by piotrmaslanka** [comment]
Failing flake8 test is intentional. I'm not importing typing for the sake of older Python's. Besides, it's trivial to fix.

**Comment by piotrmaslanka** [comment]
Besides, it's deployed on a production-grade system of mine and it's working without a hitch 💯

**Comment by tintoy** [comment]
Hey - haven’t forgotten about this, just haven’t had time to look at it yet! Work has been a little crazy this week so unfortunately I may not have time to look at it until Saturday but it will be top of my list then :)

File: seqlog/__init__.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
 from seqlog.structured_logging import set_global_log_properties as _set_global_log_properties
 from seqlog.structured_logging import clear_global_log_properties as _clear_global_log_properties
 from seqlog.structured_logging import reset_global_log_properties as _reset_global_log_properties
+from seqlog.structured_logging import set_callback_on_failure
```
Comment (tintoy) [inline seqlog/__init__.py:13]: Hi - is this import needed?

**Review by tintoy — APPROVED** [review]
Hi - this looks good to me.

If that import isn't needed then I'll remove it to make flake8 stop complaining 😉 

PS. Sorry for the delay, it's been a very busy week!

File: seqlog/__init__.py (hunk 1 above)
Comment (piotrmaslanka) [reply seqlog/__init__.py:13]: Not really.

**Comment by piotrmaslanka** [comment]
@tintoy so when are you going to release this fix on PyPI?

**Comment by tintoy** [comment]
Oh dear, I thought I had! I’ll sort that out today; sorry about that :)

**Comment by tintoy** [comment]
Published [v0.3.20](https://pypi.org/project/seqlog/0.3.20/).

**Pull request merged**